### PR TITLE
Better logic to check for new stream file in the live reader to reduce l...

### PR DESCRIPTION
**Pull request by chtyim**
...atency when new file already exists.

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/LiveStreamFileReader.java [hunk 1]
```diff
@@ -60,14 +61,14 @@ public LiveStreamFileReader(StreamConfig streamConfig, StreamFileOffset beginOff
...
+   * @param streamConfig the stream configuration.
+   * @param beginOffset the offset information to begin with.
+   * @param maxFileCheckInterval maximum interval in milliseconds for checking for new stream file.
    */
-  public LiveStreamFileReader(StreamConfig streamConfig, StreamFileOffset beginOffset, long newFileCheckInterval) {
+  public LiveStreamFileReader(StreamConfig streamConfig, StreamFileOffset beginOffset, long maxFileCheckInterval) {
     this.streamConfig = streamConfig;
     this.beginOffset = beginOffset;
-    this.newFileCheckInterval = newFileCheckInterval;
+    this.maxFileCheckInterval = maxFileCheckInterval;
```
Comment (ghelmling) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/LiveStreamFileReader.java]: Might want to add a sanity check that maxFileCheckInterval is not negative

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/LiveStreamFileReader.java [hunk 2]
```diff
@@ -110,12 +108,29 @@ public LiveStreamFileReader(StreamConfig streamConfig, StreamFileOffset beginOff
...
   }
 
   /**
+   * Computes the new file check interval based on number of retries that a new file hasn't been found.
+   */
+  private long getCheckInterval(int retries) {
+    retries++;
+    int multiplier = retries > Integer.SIZE ? Integer.MAX_VALUE : (1 << (retries - 1));
+    long interval = Math.min(100 * multiplier, maxFileCheckInterval);
+    return interval < 0 ? maxFileCheckInterval : interval;
```
Comment (ghelmling) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/LiveStreamFileReader.java]: should be interval <= 0 i think

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/LiveStreamFileReader.java (hunk 1 above)
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/LiveStreamFileReader.java]: Added a condition.

File: cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/LiveStreamFileReader.java (hunk 2 above)
Comment (chtyim) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/LiveStreamFileReader.java]: I updated the calculation. Please see if that's better.

**Comment by ghelmling** [comment]
lgtm

**Pull request merged**